### ISSUE TITLE
Assertion failed: (gnutls_certificate_set_x509_system_trust(xcred))>=0, file tcp.c, line 354, function tcp_tls_connect

### ISSUE THREAD

**Issue by Mno-hime** [rdesktop/rdesktop#332]
I built aa5164ede47d4585d29d28e2b71a0544a7b674b0 on OpenIndiana 2019.04 (illumos distribution) with GNU TLS 3.5.14 and `rdesktop -u Administrator -p xxxx XXXXXXXXXXXXXXXX` command against WS 2016 fails with:

```
Assertion failed: (gnutls_certificate_set_x509_system_trust(xcred))>=0, file tcp.c, line 354, function tcp_tls_connect
```

`rdesktop` 1.8.3 works fine.

Logs:

```
$ RDESKTOP_DEBUG=All rdesktop -u Administrator -p xxxx XXXXXXXXXXXXXXXX
Autoselecting keyboard map 'en-us' from locale
UI(debug): RDP depth: 24, display depth: 24, display bpp: 32, X server BE: 0, host BE: 0

Keyboard(debug): add_to_keymap(), adding translation, keysym=0xffe2, scancode=0x36, modifiers=0x0
Keyboard(debug): add_to_keymap(), adding translation, keysym=0xffe1, scancode=0x2a, modifiers=0x0
...
Keyboard(debug): add_to_keymap(), adding translation, keysym=0x2f, scancode=0x35, modifiers=0x0
Keyboard(debug): add_to_keymap(), adding translation, keysym=0x3f, scancode=0x35, modifiers=0x2
is_wm_active(): WM name: GNOM
Core(debug): dvc_channels_add(), Added hash=67f73722, channel_id=-1, name=Microsoft::Windows::RDS::DisplayControl, handler=47b2f0
Protocol(debug): rdp_reset_state()
Protocol(debug): mcs_connect_start()
Connecting to server using NLA...
Core(debug): tcp_connect(), trying XXXXXXXXXXXXXXXX (XXX.XXX.XXX.XXX)
Assertion failed: (gnutls_certificate_set_x509_system_trust(xcred))>=0, file tcp.c, line 354, function tcp_tls_connect
Abort (core dumped)
```

```
Loading modules: [ libc.so.1 ld.so.1 ]
> ::status
debugging core file of rdesktop (64-bit) from oi
file: /usr/bin/rdesktop
initial argv: rdesktop -v -u Administrator -p xxxx XXXXXXXXXXXXXXXX
threading model: native threads
status: process terminated by SIGABRT (Abort), pid=5166 uid=101 code=-1
panic message: Assertion failed: (gnutls_certificate_set_x509_system_trust(xcred))>=0, file tcp.c, line 354, function tcp_tls_connect
> $C
fffffd7fffdf0510 libc.so.1`_lwp_kill+0xa()
fffffd7fffdf0540 libc.so.1`raise+0x20(6)
fffffd7fffdf0590 libc.so.1`abort+0x98()
fffffd7fffdf0800 0xfffffd7feeec1e99()
fffffd7fffdf0830 tcp_tls_connect+0x239()
fffffd7fffdf09b0 cssp_connect+0x138()
fffffd7fffdf0a10 iso_connect+0x617()
fffffd7fffdf0a60 sec_connect+0x18()
fffffd7fffdf0b00 rdp_connect+0x3f()
fffffd7fffdf1020 main+0xd72()
fffffd7fffdf1050 _start_crt+0x83()
fffffd7fffdf1060 _start+0x18()
```

```
core 'core' of 5166:	rdesktop -v -u Administrator -p xxxx XXXXXXXXXXXXXXXX
0000000000400000        600K r-x--  /usr/bin/rdesktop
00000000004A6000          8K rw---  /usr/bin/rdesktop
00000000004A8000       1696K rw---  /usr/bin/rdesktop
000000000103C000        352K rw---    [ heap ]
FFFFFD7FE7A80000       1680K r-x--  /usr/lib/amd64/libunistring.so.2.1.0
FFFFFD7FE7C34000         16K rw---  /usr/lib/amd64/libunistring.so.2.1.0
FFFFFD7FEE70C000        336K r-x--  /lib/amd64/ld.so.1
FFFFFD7FEE770000         12K rwx--  /lib/amd64/ld.so.1
FFFFFD7FEE773000          8K rwx--  /lib/amd64/ld.so.1
FFFFFD7FEE780000        160K r-x--  /usr/lib/amd64/libxcb.so.1.1.0
FFFFFD7FEE7B8000          4K rw---  /usr/lib/amd64/libxcb.so.1.1.0
FFFFFD7FEE7D0000         68K r-x--  /lib/amd64/libsocket.so.1
FFFFFD7FEE7F1000          4K rw---  /lib/amd64/libsocket.so.1
FFFFFD7FEE830000         12K r-x--  /usr/lib/amd64/libXau.so.6.0.0
FFFFFD7FEE843000          4K rw---  /usr/lib/amd64/libXau.so.6.0.0
FFFFFD7FEE850000          4K rwx--    [ anon ]
FFFFFD7FEE870000         64K rwx--    [ anon ]
FFFFFD7FEE890000       1688K r-x--  /usr/lib/amd64/gnutls-3/libgnutls.so.30.14.6
FFFFFD7FEEA46000         56K rw---  /usr/lib/amd64/gnutls-3/libgnutls.so.30.14.6
FFFFFD7FEEA54000          4K rw---  /usr/lib/amd64/gnutls-3/libgnutls.so.30.14.6
FFFFFD7FEEA70000       1296K r-x--  /usr/lib/amd64/libX11.so.4.0.0
FFFFFD7FEEBC4000         16K rw---  /usr/lib/amd64/libX11.so.4.0.0
FFFFFD7FEEBD0000       1484K r-x--  /usr/lib/amd64/libp11-kit.so.0.3.0
FFFFFD7FEED53000         80K rw---  /usr/lib/amd64/libp11-kit.so.0.3.0
FFFFFD7FEED80000        528K r-x--  /lib/amd64/libnsl.so.1
FFFFFD7FEEE14000         12K rw---  /lib/amd64/libnsl.so.1
FFFFFD7FEEE17000         32K rw---  /lib/amd64/libnsl.so.1
FFFFFD7FEEE30000       1552K r-x--  /lib/amd64/libc.so.1
FFFFFD7FEEFC4000         48K rw---  /lib/amd64/libc.so.1
FFFFFD7FEEFD0000         16K rw---  /lib/amd64/libc.so.1
FFFFFD7FEEFE0000          4K rwx--    [ anon ]
FFFFFD7FEF000000          4K rwx--    [ anon ]
FFFFFD7FEF020000          4K rwx--    [ anon ]
FFFFFD7FEF040000        128K r-x--  /usr/lib/amd64/libgss.so.1
FFFFFD7FEF060000          4K rw---  /usr/lib/amd64/libgss.so.1
FFFFFD7FEF061000          4K rw---  /usr/lib/amd64/libgss.so.1
FFFFFD7FEF080000          4K rwx--    [ anon ]
FFFFFD7FEF090000          4K rwx--    [ anon ]
FFFFFD7FEF0A0000         24K rwx--    [ anon ]
FFFFFD7FEF0C0000        120K r-x--  /usr/lib/amd64/libXext.so.0.0.0
FFFFFD7FEF0EE000          4K rw---  /usr/lib/amd64/libXext.so.0.0.0
FFFFFD7FEF100000          4K rwx--    [ anon ]
FFFFFD7FEF110000          4K rwx--    [ anon ]
FFFFFD7FEF140000          4K rwx--    [ anon ]
FFFFFD7FEF160000          4K rwx--    [ anon ]
FFFFFD7FEF170000          4K rwx--    [ anon ]
FFFFFD7FEF180000         64K rwx--    [ anon ]
FFFFFD7FEF1A0000          4K r-x--  /lib/amd64/libintl.so.1
FFFFFD7FEF1C0000        128K rwx--    [ anon ]
FFFFFD7FEF1E2000         80K r----    [ anon ]
FFFFFD7FEF200000          4K rw---    [ anon ]
FFFFFD7FEF220000        268K r-x--  /usr/lib/amd64/libnettle.so.6.3
FFFFFD7FEF273000          8K rw---  /usr/lib/amd64/libnettle.so.6.3
FFFFFD7FEF290000          4K rw---    [ anon ]
FFFFFD7FEF2B0000          4K rwx--    [ anon ]
FFFFFD7FEF2D0000          4K rwx--    [ anon ]
FFFFFD7FEF2F0000          4K rwx--    [ anon ]
FFFFFD7FEF300000        132K r-x--  /usr/lib/amd64/libidn2.so.0.3.5
FFFFFD7FEF331000          4K rw---  /usr/lib/amd64/libidn2.so.0.3.5
FFFFFD7FEF350000          4K rwx--    [ anon ]
FFFFFD7FEF360000         36K r-x--  /usr/lib/amd64/libffi.so.6.0.4
FFFFFD7FEF379000          4K rw---  /usr/lib/amd64/libffi.so.6.0.4
FFFFFD7FEF390000          4K rwx--    [ anon ]
FFFFFD7FEF3A0000          4K rwx--    [ anon ]
FFFFFD7FEF3C0000          4K rwx--    [ anon ]
FFFFFD7FEF3E0000          4K rwx--    [ anon ]
FFFFFD7FEF3F0000          4K r----*   [ anon ]
FFFFFD7FFFDEE000         72K rw---    [ stack ]
         total        13008K
```

**Comment by CendioOssman** [comment]
So an assert() is perhaps a bit excessive here, but it basically means it could not load the system CA database. That in turn means that certificates from well known CAs (e.g. GoDaddy) won't be automatically trusted.

I'll have a luck at making this just a warning.

**Comment by CendioOssman** [comment]
Should be fixed in 6028c999ba7852d560b7831853502e675c65709a.
